plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TypeConverter.java

```diff
@@ -112,7 +113,7 @@ public static Type toTrinoType(org.apache.iceberg.types.Type type, TypeManager t
         throw new UnsupportedOperationException(format("Cannot convert from Iceberg type '%s' (%s) to Trino type", type, type.typeId()));
     }
 
-    public static org.apache.iceberg.types.Type toIcebergType(Type type)
+    public static org.apache.iceberg.types.Type toIcebergType(Type type, Optional<ColumnIdentity> columnIdentity)
```

Comment:
Can we split the public part of this API into two methods:

- `toIcebergType(Type, ColumnIdentity)`
- `icebergTypeForNewColumn(Type, AtomicInteger nextFieldId)`

The private methods methods can use Optionals like you have, but that makes it clear that an empty column identity should only be used for columns that don't exist yet

Reply:
@alexjo2144 I finished a version, please help check it out.